controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/ScaleOperationTask.java

```diff
@@ -83,31 +85,58 @@ public ScaleOperationTask(final StreamMetadataTasks streamMetadataTasks,
     }
 
     @VisibleForTesting
-    public CompletableFuture<EpochTransitionRecord> runScale(ScaleOpEvent scaleInput, boolean runOnlyIfStarted, OperationContext context, String delegationToken) { // called upon event read from requeststream
+    public CompletableFuture<Void> runScale(ScaleOpEvent scaleInput, boolean isManualScale, OperationContext context, String delegationToken) { // called upon event read from requeststream
         String scope = scaleInput.getScope();
         String stream = scaleInput.getStream();
-        // create epoch transition node (metadatastore.startScale)
-        // Note: if we crash before deleting epoch transition, then in rerun (retry) it will be rendered inconsistent
+        // Note: There are two interesting cases for retry of partially completed workflow that are interesting:
+        // if we crash before resetting epoch transition (complete scale step), then in rerun ETR will be rendered inconsistent
         // and deleted in startScale method.
```

Comment:
Is this observation about deleting the the inconsistent ETR referring to this line?

https://github.com/pravega/pravega/pull/2971/files#diff-5cfebc728ee7842a929102d84860dfa8R504

I believe we are changing the name of the method to `submitScale` and the method to validate is also changing. Should we update this comment?

Reply:
> Is this observation about deleting the the inconsistent ETR referring to this line?
> https://github.com/pravega/pravega/pull/2971/files#diff-5cfebc728ee7842a929102d84860dfa8R504
No, its referring to this line 630 (new file)

> I believe we are changing the name of the method to submitScale and the method to validate is also changing. Should we update this comment?
Actually we broke the logic in original `startScale` method into two different methods - `submitScale` and `startScale`. 
So the inconsistency check is verified in `startScale`

Reply:
> No, its referring to this line 630 (new file)

Did you mean to post a link?

Reply:
https://github.com/pravega/pravega/pull/2971/files#diff-5cfebc728ee7842a929102d84860dfa8R630